github: Fix external workloads test file syntax

brb opened the pull request:
See commit msgs.

Fixes: 5726e9f75d ("workflows: add external workload conformance test")

nbusseneau commented on .github/workflows/conformance-externalworkloads.yml:
```diff
@@ -76,7 +76,7 @@ jobs:
     if: |
       (github.event_name == 'issue_comment' && (
         startsWith(github.event.comment.body, 'ci-external-workloads') ||
-        (startsWith(github.event.comment.body, 'test-me-please'))
```
This is correct and can be left as-is (consistent with other workflows).

nbusseneau commented on .github/workflows/conformance-externalworkloads-v1.10.yml:
```diff
@@ -72,8 +72,8 @@ jobs:
-        (startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
-        startsWith(github.event.comment.body, 'test-backport-1.10')) &&
+        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
+        startsWith(github.event.comment.body, 'test-backport-1.10')
```
This should be:
```suggestion
        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
        (startsWith(github.event.comment.body, 'test-backport-1.10'))
```
I am joining a patch with this change and another one.

pchaigno replied:
Why? Does it make a difference?

nbusseneau replied:
No, it's just to be consistent with the other workflows (just like for the other comment above). The failure was indeed related to the `&&`, but GH suggestion shows the diff w.r.t. to the new code rather than the original one.

nbusseneau reviewed (changes requested):
Thanks for catching this. I forgot a `git add`... 🤦🏻 

Can you replace your changes with the following instead? These are the correct changes. Thanks :)

```
From 78427044d56f0c9bc5debabf8458729da431a43a Mon Sep 17 00:00:00 2001
From: Nicolas Busseneau <nicolas@isovalent.com>
Date: Fri, 30 Jul 2021 15:52:54 +0200
Subject: [PATCH] workflows: fix external workloads 1.10 YAML

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
---
 .github/workflows/conformance-externalworkloads-v1.10.yml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/.github/workflows/conformance-externalworkloads-v1.10.yml b/.github/workflows/conformance-externalworkloads-v1.10.yml
index fd55279a9..768a0bb99 100644
--- a/.github/workflows/conformance-externalworkloads-v1.10.yml
+++ b/.github/workflows/conformance-externalworkloads-v1.10.yml
@@ -72,8 +72,8 @@ jobs:
     name: Deduce required tests from code changes
     if: |
       (github.event_name == 'issue_comment' && (
-        (startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
-        startsWith(github.event.comment.body, 'test-backport-1.10')) &&
+        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
+        (startsWith(github.event.comment.body, 'test-backport-1.10'))
       )) ||
       (github.event_name == 'schedule' && github.repository == 'cilium/cilium') ||
       github.event_name == 'pull_request'
@@ -112,8 +112,8 @@ jobs:
     needs: check_changes
     if: |
       (github.event_name == 'issue_comment' && (
-        startsWith(github.event.comment.body, 'ci-external-workloads') ||
-        (startsWith(github.event.comment.body, 'test-me-please') && (needs.check_changes.outputs.tested == 'true'))
+        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
+        (startsWith(github.event.comment.body, 'test-backport-1.10') && (needs.check_changes.outputs.tested == 'true'))
       )) ||
       (github.event_name == 'schedule' && github.repository == 'cilium/cilium') ||
       github.event_name == 'pull_request'
-- 
2.32.0
```

brb committed:
github: Fix external workloads test files

The GH action runner reported:

    The workflow is not valid.
    .github/workflows/conformance-externalworkloads-v1.10.yml (Line: 73,
    Col: 9): Unexpected symbol: ')'. Located at position 185 within
    expression: (github.event_name == 'issue_comment' && (
      (startsWith(github.event.comment.body, 'ci-external-workloads-v1.10')
      ||
        startsWith(github.event.comment.body, 'test-backport-1.10')) &&
        )) ||
        (github.event_name == 'schedule' && github.repository ==
        'cilium/cilium') ||
        github.event_name == 'pull_request'

Fixes: 5726e9f75d ("workflows: add external workload conformance test")
Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Martynas Pumputis <m@lambda.lt>

nbusseneau commented on .github/workflows/conformance-externalworkloads-v1.10.yml:
```diff
@@ -112,8 +112,8 @@ jobs:
       (github.event_name == 'issue_comment' && (
-        startsWith(github.event.comment.body, 'ci-external-workloads') ||
-        (startsWith(github.event.comment.body, 'test-me-please') && (needs.check_changes.outputs.tested == 'true'))
+        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
+        (startsWith(github.event.comment.body, 'test-backport-1.10') && (needs.check_changes.outputs.tested == 'true'))
```
👍🏻

nbusseneau commented on .github/workflows/conformance-externalworkloads-v1.10.yml:
```diff
@@ -72,8 +72,8 @@ jobs:
-        (startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
-        startsWith(github.event.comment.body, 'test-backport-1.10')) &&
+		startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
+        (startsWith(github.event.comment.body, 'test-backport-1.10') && (needs.check_changes.outputs.tested == 'true'))
```
Think the `(needs.check_changes.outputs.tested == 'true')` incorrectly sneaked in there:
```suggestion
        startsWith(github.event.comment.body, 'ci-external-workloads-v1.10') ||
        (startsWith(github.event.comment.body, 'test-backport-1.10'))
```

nbusseneau reviewed (approved):
🚀
